src/gui/guiFormSpecMenu.cpp

```diff
@@ -1911,35 +1903,44 @@ void GUIFormSpecMenu::parseImageButton(parserData* data, const std::string &elem
 		if (type == "image_button_exit")
 			spec.is_exit = true;
 
-		GUIButtonImage *e = GUIButtonImage::addButton(Environment, rect, this, spec.fid, spec.flabel.c_str());
+		GUIButtonImage *e = GUIButtonImage::addButton(Environment, rect, m_tsrc,
+				this, spec.fid, spec.flabel.c_str());
 
 		if (spec.fname == data->focused_fieldname) {
 			Environment->setFocus(e);
 		}
 
-		auto style = getStyleForElement("image_button", spec.fname);
-		e->setFromStyle(style, m_tsrc);
-
-		// We explicitly handle these arguments *after* the style properties in
-		// order to override them if they are provided
-		if (!image_name.empty())
-		{
-			video::ITexture *texture = m_tsrc->getTexture(image_name);
-			e->setForegroundImage(guiScalingImageButton(
-				Environment->getVideoDriver(), texture, geom.X, geom.Y));
+		// Push temporary styles with parameters
```

Comment:
Rather than pushing temporary styles, I wonder if it would be better to copy the results of `getStyleForElement` and then push these values to it?

Reply:
Yeah, good call. I think this wasn't possible when I first started the PR, and I just forgot to simplify things after that changed.